sale_loyalty_criteria_order_based/models/loyalty_program.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2022 Ooops404
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import ast
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.osv.expression import expression
+
+
+class LoyaltyProgram(models.Model):
+    _inherit = "loyalty.program"
+
+    rule_order_domain = fields.Char(string="Based on Order", default="[]")
```

Comment:
Wouldn't it make more sense to use a M2o on `ir.filter` in order to make it reusable between different programs?